nlp4all/models/Project.py

```diff
@@ -1,29 +1,35 @@
 """Project Model""" # pylint: disable=invalid-name
 
+from __future__ import annotations
+
 from random import sample
-from sqlalchemy import Column, Integer, String, ForeignKey, JSON
-from sqlalchemy.orm import relationship, load_only
+from sqlalchemy import Integer, String, ForeignKey
+from sqlalchemy.orm import relationship, load_only, Mapped, mapped_column
+
+from .database import Base, project_data_source_table, project_categories_table, MutableJSON
 
-from .database import Base
-from . import Tweet
+from . import DataSource, BayesianAnalysis, DataTagCategory, Data
 
 
 class Project(Base):
     """Project model."""
 
     __tablename__ = "project"
-    id = Column(Integer, primary_key=True)
-    name = Column(String(50))
-    description = Column(String)
-    organization = Column(Integer, ForeignKey("organization.id"))
-    analyses = relationship("BayesianAnalysis")
-    categories = relationship(
-        "TweetTagCategory",
-        secondary="project_categories",
+    id: Mapped[int] = mapped_column(primary_key=True)
+    name: Mapped[str] = mapped_column(String(50))
+    description: Mapped[str] = mapped_column()
+    organization: Mapped[int] = mapped_column(Integer, ForeignKey("organization.id"))
+    analyses: Mapped[list[BayesianAnalysis]] = relationship()
+    categories: Mapped[list[DataTagCategory]] = relationship(
+        secondary=project_categories_table,
+        back_populates="projects")
+    data_sources: Mapped[list[DataSource]] = relationship(
```

Comment:
for now, (with the possibility of future expansion) this should be a single data source, to map to the old behaviour of tweets assigned to a project...

this means you could directly replace Project -> tweets with Project->data_source->data